### DISCUSSION TITLE
Configuration test without startup?

### DISCUSSION THREAD

**Discussion by yaleman** [coredns/coredns#5955] [Q&A] [answered]
I'd like to be able to add a step to the CI in my git-managed-config where I run the equivalent of `coredns -conf Corefile -check` and check for errors, but I don't want it to actually start the server and continue to run.

Is this possible? It's the kind of thing I do with other applications, and I can manage it otherwise, but it'd be handy to have.

**Comment by chrisohaver** [comment, marked as answer]
Definitely would be useful, but harder than it sounds to actually implement.  See the most recent open enhancement request: https://github.com/coredns/coredns/issues/5152

**Reply by yaleman** [reply]
Thanks, I tried searching a variety of things in issues and couldn't find it (yay, generic terms). I'll watch that space.
